client/landing/jetpack-cloud/components/filter-bar/index.tsx

```diff
@@ -0,0 +1,159 @@
+/**
+ * External dependencies
+ */
+import { useSelector } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+import React, { FunctionComponent, useEffect, useState, useRef } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '@automattic/components';
+import { getHttpData } from 'state/data-layer/http-data';
+import {
+	getRequestActivityActionTypeCountsId,
+	requestActivityActionTypeCounts,
+} from 'state/data-getters';
+import ActivityTypeSelector from './activity-type-selector';
+import Gridicon from 'components/gridicon';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface Filter {
+	group?: string[];
+	after?: string;
+	before?: string;
+	page: number;
+}
+
+interface ActivityCount {
+	count: number;
+	key: string;
+	name: string;
+}
+
+interface Props {
+	siteId: number;
+	filter: Filter;
+	onFilterChange: ( newFilter: Filter ) => void;
+	showActivityTypeSelector?: boolean;
+	showDateRangeSelector?: boolean;
+}
+
+const FilterBar: FunctionComponent< Props > = ( {
+	siteId,
+	filter,
+	showActivityTypeSelector = true,
+	showDateRangeSelector = true,
+	onFilterChange,
+} ) => {
+	const translate = useTranslate();
+	const [ isActivityTypeSelectorVisible, setIsActivityTypeSelectorVisible ] = useState( false );
+	const [ isDateRangeSelectorVisible, setIsDateRangeSelectorVisible ] = useState( false );
+
+	const activityActionTypeCounts = useSelector< object, ActivityCount[] >(
+		() => getHttpData( getRequestActivityActionTypeCountsId( siteId, filter ) ).data
+	);
+
+	const toggleIsActivityTypeSelectorVisible = () => {
+		setIsDateRangeSelectorVisible( false );
+		setIsActivityTypeSelectorVisible( ! isActivityTypeSelectorVisible );
+	};
+
+	const closeActivityTypeSelector = () => {
+		setIsActivityTypeSelectorVisible( false );
+	};
+
+	const toggleIsDateRangeSelectorVisible = () => {
+		setIsActivityTypeSelectorVisible( false );
+		setIsDateRangeSelectorVisible( ! isDateRangeSelectorVisible );
+	};
+
+	const activityTypeButtonRef = useRef< Button >( null );
+	const dateRangeButtonRef = useRef< Button >( null );
+
+	// const closeDateRangeSelector = () => {
+	// 	setIsDateRangeSelectorVisible( false );
+	// };
+
+	const getButtonClassName = ( isActive: boolean, hasSelections: boolean ) => {
+		if ( hasSelections ) {
+			return 'filter-bar__button-with-selections';
+		}
+
+		return isActive ? 'filter-bar__button-active' : 'filter-bar__button';
+	};
+
+	const renderActivityTypeSelectorButtonText = ( groups?: string[] ) =>
+		groups
+			? translate( 'Activity type: %d selected', {
+					args: [ groups.length ],
+			  } )
+			: translate( 'Activity type' );
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'Activity Type' )` *ES Score: 10*